demos/python_demos/object_detection_demo_yolov3_async/object_detection_demo_yolov3_async.py

```diff
@@ -161,44 +161,35 @@ def parse_yolo_region(blob, resized_image_shape, original_im_shape, params, thre
     orig_im_h, orig_im_w = original_im_shape
     resized_image_h, resized_image_w = resized_image_shape
     objects = list()
-    predictions = blob.flatten()
-    side_square = params.side * params.side
     size_normalizer = (resized_image_w, resized_image_h) if params.isYoloV3 else (params.side, params.side)
+    bbox_size = params.coords + 1 + params.classes
     # ------------------------------------------- Parsing YOLO Region output -------------------------------------------
-    for i in range(side_square):
-        row = i // params.side
-        col = i % params.side
-        for n in range(params.num):
-            obj_index = entry_index(params.side, params.coords, params.classes, n * side_square + i, params.coords)
-            object_probability = predictions[obj_index]
-            if object_probability < threshold:
-                continue
-            box_index = entry_index(params.side, params.coords, params.classes, n * side_square + i, 0)
-            # Network produces location predictions in absolute coordinates of feature maps.
-            # Scale it to relative coordinates.
-            x = (col + predictions[box_index + 0 * side_square]) / params.side
-            y = (row + predictions[box_index + 1 * side_square]) / params.side
-            # Value for exp is very big number in some cases so following construction is using here
-            try:
-                width = exp(predictions[box_index + 2 * side_square])
-                height = exp(predictions[box_index + 3 * side_square])
-            except OverflowError:
-                continue
-            # Depends on topology we need to normalize sizes by feature maps (up to YOLOv3) or by input shape (YOLOv3)
-            width = width * params.anchors[2 * n] / size_normalizer[0]
-            height = height * params.anchors[2 * n + 1] / size_normalizer[1]
-            class_probabilities = []
-            for j in range(params.classes):
-                class_index = entry_index(params.side, params.coords, params.classes, n * side_square + i,
-                                          params.coords + 1 + j)
-                class_probabilities.append(predictions[class_index])
-
-            class_id = np.argmax(class_probabilities)
-            confidence = class_probabilities[class_id]*object_probability
-            if confidence < threshold:
-                continue
-            objects.append(scale_bbox(x=x, y=y, height=height, width=width, class_id=class_id, confidence=confidence,
-                                      im_h=orig_im_h, im_w=orig_im_w, is_proportional=is_proportional))
+    for row, col, n in np.ndindex(params.side, params.side, params.num):
+        # Getting raw values for each detection bounding box
+        bbox = predictions[0, n*bbox_size:(n+1)*bbox_size, row, col]
+        x, y, width, height, object_probability = bbox[:5]
+        class_probabilities = bbox[5:]
```

Comment:
```suggestion
        x, y, width, height, object_probability, *class_probabilities = bbox
```

Reply:
Well, now I think, that initial code (with indexes) was much more clearer because it shows us how exactly values extracted from the blob. If anyone supports me, I'll revert this change, otherwise agree with you

Reply:
That suggestion changes `class_probabilities`'s type. It will be a list. So `bbox[5:]` will be cast to list and than back to ndarray by np.argmax (but I'm not actually sure if argmax will cast list to ndarray before processing).

The current state and the suggestion also shows how values are extracted from the array: the fist 5 elements are special with given names and the rest are class_probabilities.

Reply:
> That suggestion changes class_probabilities's type. It will be a list. So bbox[5:] will be cast to list and than back to ndarray by np.argmax (but I'm not actually sure if argmax will cast list to ndarray before processing).

Yeah, that's a good point. It would be better if it remained as an `ndarray`, if for no other reason than to be memory-efficient. I'm cancelling my suggestion. :-)

> Well, now I think, that initial code (with indexes) was much more clearer because it shows us how exactly values extracted from the blob.

I don't see how it was clearer. It obscured all accesses behind complicated index calculations. To me, it seems much clearer now.

Reply:
Exactly. The `entry_index` function left me and some of my peers at the Intel OpenVINO-related project spooked. It is very difficult to understand the way the indices are calculated from the flattened blob. 

Either the process should be documented or the function should be removed.